File: 240116/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.java
import java.util.Scanner;
import java.util.Arrays;
class Student implements Comparable<Student>{
    String name;
    int height,weight;
    public Student(String name,int height,int weight){
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
    public int compareTo(Student student){
        if (height!=student.height){
            return height - student.height;
        }
        return student.weight - weight;
    }
}
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Student[] peo = new Student[n];

        for(int i = 0 ; i<n; i++){
            String name = sc.next();
            int height = sc.nextInt();
            int weight = sc.nextInt();
            peo[i] = new Student(name, height, weight);
        }
        Arrays.sort(peo);

        for (int i = 0; i<n; i++){
            System.out.println(peo[i].name+" "+peo[i].height+" "+peo[i].weight);
        }
    }
}

File: 240116/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.py
n = int(input())

class Student:

    def __init__(self,name,height,weight):
        self.name = name
        self.height = height
        self.weight = weight
peo = []

for i in range(n):
    name,height,weight = map(str,input().split())
    height = int(height)
    weight = int(weight)

    peo.append(Student(name,height,weight))

peo.sort(key = lambda x: (x.height,-x.weight))

for i in range(n):
    print(peo[i].name,peo[i].height,peo[i].weight)